### PR TITLE
[origin] Adding openshift vesion and node information

### PR DESCRIPTION
**Pull request by sferich888**
Should help improve things for https://bugzilla.redhat.com/show_bug.cgi?id=1246423

### CODE REVIEW

**Comment by pmoravec** [comment]
Please fix pep8 formatting issue:

    sos/plugins/origin.py:154:44: E128 continuation line under-indented for visual indent

File: sos/plugins/origin.py [hunk 1]
```diff
@@ -119,6 +119,9 @@ def setup(self):
             # Note: Information about nodes, events, pods, and services
             # is already collected by the Kubernetes plugin
             self.add_cmd_output([
+                "openshift vesion",
+                "%s vesion" % self.oc_cmd,
+                "%s vesion" % self.oadm_cmd,
```
Comment (codificat) [inline sos/plugins/origin.py]: Note that these will return basically the same information.
Moreover, we have version information from the rpm and/or docker plugins, so this sounds redundant, no?

File: sos/plugins/origin.py [hunk 2]
```diff
@@ -146,6 +149,13 @@ def setup(self):
                 self.add_cmd_output("%s logs -n default %s"
                                     % (self.oc_cmd, pod))
 
+            # get a node description from the cluster ndoes
```
Comment (codificat) [inline sos/plugins/origin.py]: I believe that this information belongs to the kubernetes plugin.
That plugin already collects node information - not a describe at the moment, but in any case I believe that if we need to amend the information we collect from nodes it should happen in the kubernetes plugin instead.

**Comment by codificat** [comment]
@sferich888 I believe that this PR is a bit redundant with information collected by other plugins. Is there a reason we should collect that information explicitly in the origin plugin?

File: sos/plugins/origin.py (hunk 1 above)
Comment (sferich888) [inline sos/plugins/origin.py]: Yes, however the version information collected by this is for user convenience (to the user looking at the data), I will remove some of the duplication.

File: sos/plugins/origin.py (hunk 2 above)
Comment (sferich888) [inline sos/plugins/origin.py]: It is, but its not in the most "readable" format, or separated in a way that is useful to an admin looking at the system. 

Example: 
```
{
    "kind": "Event",
    "apiVersion": "v1",
    "metadata": {
        "name": "docker-registry-7-5h9cg.146c505f7341377c",
        "namespace": "default",
        "selfLink": "/api/v1/namespaces/default/events/docker-registry-7-5h9cg.146c505f7341377c",
        "uid": "e8d4e039-664d-11e6-977b-fa163e71e8b9",
        "resourceVersion": "2574470",
        "creationTimestamp": "2016-08-19T20:45:48Z",
        "deletionTimestamp": "2016-08-22T15:40:46Z"
    },
    "involvedObject": {
        "kind": "Pod",
        "namespace": "default",
        "name": "docker-registry-7-5h9cg",
        "uid": "5574d66d-3d76-11e6-9648-fa163e71e8b9",
        "apiVersion": "v1",
        "resourceVersion": "2366618",
        "fieldPath": "spec.containers{registry}"
    },
    "reason": "Pulled",
    "message": "Container image \"registry.access.redhat.com/openshift3/ose-docker-registry:v3.2.0.46\" already present on machine",
    "source": {
        "component": "kubelet",
        "host": "node2.vault.com"
    },
    "firstTimestamp": "2016-08-19T20:45:49Z",
    "lastTimestamp": "2016-08-22T13:40:46Z",
    "count": 1125,
    "type": "Normal"
}
```
In short events are stored in "namespaces" and tied to nodes, so you would need to parse the json from the _kubernetes plugin_ to get this data viewable in a form where it is useful.

**Comment by sferich888** [comment]
I have made the suggested changes, and feel this meets everyone in the middle.

File: sos/plugins/kubernetes.py [hunk 3]
```diff
@@ -88,6 +88,13 @@ def setup(self):
                 "{} get -o json pv".format(kube_cmd)
```
Comment (bmr-cymru) [inline sos/plugins/kubernetes.py]: Needs a "[plugin] change description" commit format, Signed-off-by, and there's a little tpyo in the text ('convienacne').

File: sos/plugins/kubernetes.py [hunk 4]
```diff
@@ -88,6 +88,13 @@ def setup(self):
...
             ])
 
+            # get a node description from ndoes (convenience)
+            nodes = self.get_command_output("%s get nodes -o name"
+                                            % self.oc_cmd)
+            for node in nodes['output'].splitlines():
```
Comment (bmr-cymru) [inline sos/plugins/kubernetes.py]: Adding things "for convenience" is a common red-flag in sos; strictly, it is still primarily a tool for support data collection. That means that user convenience (for the support engineers viewing the data) comes second to requirements like:

* robustness
* space consumption
* time consumption

The user whose convenience we are most concerned with is the end-customer who is having to run sos in order to get assistance from a vendor or project.

That said, I don't have a flat-out rule on this, unless there are obvious problems for other users (unreasonably large data sets, runtime etc.

So for this I think I am happy so long as there is agreement among the maintainers of the related pieces; you guys are an active bunch and these plugins and the projects they relate to are all fairly quick-moving at the moment.

So long as we aren't making burdens for Regular Joe end-users then I'm happy with these additions.

Comment (codificat) [inline sos/plugins/kubernetes.py]: I agree that these sound redundant/unnecessary - but also they shouldn't imply a major burden for the end user (the node describe would add ~2k / ~0.5s per node), so , so I'm also fine to add them if people find them useful.

Maybe @TurboTurtle has comments about the Kuberentes plugin change. To begin with, you should replace invocations to `oc` with invocations of `kubectl`; also you might just replace the `get nodes` + `for ... describe` loop with a single `kubectl describe node` which would get them all.

**Comment by bmr-cymru** [comment]
@sferich888 hey, this patch just needs the fixes for [Contribution Guidelines](https://github.com/sosreport/sos/wiki/Contribution-Guidelines) mentioned back in Sept. to get merged - missing Signed-off-by, and a typo ('get a node description from ndoes').

GitHub is still showing me the branch with commit 36abc1c at HEAD ('[origin] Adding openshift vesion and node information') from August 22.

**Comment by bmr-cymru** [comment]
We will be releasing 3.4 next month: I would like to include this change but the PR has not been updated to reflect review comments since August last year.

Please push an updated HEAD for this branch if you would like to see this included in the release.

**Comment by bmr-cymru** [comment]
@codificat @sferich888 any update on this PR? It's been hanging around a long time and missed 3.4 as it was not updated in time. It seems that some of the changes are still relevant (e.g. we don't collect `openshift version` anywhere).

**Comment by codificat** [comment]
I'll leave it to @sferich888 to decide if the PR is still relevant or if it can be closed.

From my point of view I think we already collect the information suggested here: openshift version can be seen in the installed RPMs or container(s) (collected by the `rpm` and `docker` plugins), and the information that `oc describe node` summarizes should also be available in other existing collection points.

If we decide to include the changes suggested here, there's one needed fix I could spot: `k_cmd` instead of `oc_cmd` in the kubernetes plugin.

**Comment by pmoravec** [comment]
Bump to @sferich888 : is the PR still relevant?

**Comment by sferich888** [comment]
relevant yes! Needed no.

**Pull request closed without merging**